src/protocols/rdp/settings.c

```diff
@@ -1403,6 +1415,8 @@ void guac_rdp_push_settings(guac_client* client,
     rdp_settings->DesktopHeight = guac_settings->height;
     rdp_settings->AlternateShell = guac_strdup(guac_settings->initial_program);
     rdp_settings->KeyboardLayout = guac_settings->server_layout->freerdp_keyboard_layout;
+    rdp_settings->KeyboardCodePage = guac_settings->server_language;
```

Comment:
@richlanc 

How did you match the rdp locale codes found in [freerdp](https://pub.freerdp.com/api/locale_8h.html) that we are sending as the `server_language` parameter to this `KeyboardCodePage` parameter.

The documentation on the code page says:

> A 32-bit, unsigned integer. If the flags field does not contain the INFO_UNICODE flag (0x00000010), then this field MUST contain the ANSI code page descriptor being used by the client (for a list of code pages, see [[MSDN-CP]](https://go.microsoft.com/fwlink/?LinkId=89981)) to encode the character fields in the Info Packet and Extended Info Packet (section [2.2.1.11.1.1.1](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/05ada9e4-a468-494b-8694-eb806a0ecc89)). However, if the flags field contains the INFO_UNICODE flag, then the CodePage field MUST contain the active language identifier in the low-word[<16>](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/cbe1ed0a-d320-4ea5-be5a-f2eb6e032853#Appendix_A_16) (for a list of language identifiers, see [[MSDN-MUI]](https://go.microsoft.com/fwlink/?LinkId=90048));

So I presume we are looking at the latter section and we can be sure that guac-server will handle this INFO_UNICODE when it interfaces with rdp.

So going to the link for the active language  identifier in low word: 

> [<16> Section 2.2.1.11.1.1](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/732394f5-e2b5-4ac5-8a0a-35345386b0d1#Appendix_A_Target_16): The Microsoft RDP 6.0 client incorrectly sends the active input locale identifier in the CodePage field of the Info Packet structure.

I am not sure what to make of it?


Furthermore If I go to the 

> for a list of language identifiers, see [[MSDN-MUI]](https://go.microsoft.com/fwlink/?LinkId=90048))

I find a banner saying that language identifiers are deperecated and discouraged.